### PR TITLE
docs: add docs for security group rules for environment's security group

### PR DESCRIPTION
**Pull request by paragbhingre**
Related to:  #3810

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: site/content/docs/manifest/environment.en.md [hunk 1]
```diff
@@ -145,6 +145,33 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
 The Availability Zone name assigned to the subnet. The `az` field is optional, by default Availability Zones are assigned in alphabetical order.
 This field is mutually exclusive with `id`.
 
+<span class="parent-field">network.vpc.</span><a id="network-vpc-security-group" href="#network-vpc-security-group" class="field">`security_group`</a> <span class="type">Map</span>  
+Configure security group rules to add to the environment's security group.
```
Comment (huanjani) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
Rules of the environment's security group.
```

File: site/content/docs/manifest/environment.en.md [hunk 2]
```diff
@@ -145,6 +145,33 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+
+<span class="parent-field">network.vpc.security_group</span><a id="network-vpc-security-group-ingress" href="#network-vpc-security-group-ingress" class="field">`ingress`</a> <span class="type">Map</span>  
+Ingress rules to allow for the environment's security group.
```
Comment (huanjani) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
Rules controlling inbound traffic.
```

File: site/content/docs/manifest/environment.en.md [hunk 3]
```diff
@@ -145,6 +145,33 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+  vpc:
+    security_group:
+      ingress:
+        - ip_protocol: tcp
+          ports: 0-65535  
+          cidr: 0.0.0.0/0
+```
+
+<span class="parent-field">network.vpc.security_group</span><a id="network-vpc-security-group-egress" href="#network-vpc-security-group-egress" class="field">`egress`</a> <span class="type">Map</span>  
+Egress rules to allow for the environment's security group.
```
Comment (huanjani) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
Rules controlling outbound traffic.
```

File: site/content/docs/manifest/environment.en.md [hunk 4]
```diff
@@ -145,6 +145,33 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+Rules controlling outbound traffic.
+```yaml
+network:
+  vpc:
+    security_group:
+      egress:
+        - ip_protocol: tcp
+          ports: 80
+          cidr: 0.0.0.0/0
+```
```
Comment (Lou1415926) [inline site/content/docs/manifest/environment.en.md:194]: We probably need to include a section for each of the fields `ip_protocol`, `ports` and `cidr`. For example: https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#publish-topics-topic

File: site/content/docs/manifest/environment.en.md [hunk 5]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
 The Availability Zone name assigned to the subnet. The `az` field is optional, by default Availability Zones are assigned in alphabetical order.
 This field is mutually exclusive with `id`.
 
+<span class="parent-field">network.vpc.</span><a id="network-vpc-security-group" href="#network-vpc-security-group" class="field">`security_group`</a> <span class="type">Map</span>  
+Rules of the environment's security group.
+
+<span class="parent-field">network.vpc.security_group</span><a id="network-vpc-security-group-ingress" href="#network-vpc-security-group-ingress" class="field">`ingress`</a> <span class="type">Map</span>
```
Comment (Lou1415926) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
<span class="parent-field">network.vpc.security_group.</span><a id="network-vpc-security-group-ingress" href="#network-vpc-security-group-ingress" class="field">`ingress`</a> <span class="type">Map</span>  
```

File: site/content/docs/manifest/environment.en.md [hunk 6]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+network:
+  vpc:
+    security_group:
+      ingress:
+        - ip_protocol: tcp
+          ports: 0-65535  
+          cidr: 0.0.0.0/0
+```
+
+<span class="parent-field">network.vpc.security_group</span><a id="network-vpc-security-group-egress" href="#network-vpc-security-group-egress" class="field">`egress`</a> <span class="type">Map</span>
```
Comment (Lou1415926) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
<span class="parent-field">network.vpc.security_group.</span><a id="network-vpc-security-group-egress" href="#network-vpc-security-group-egress" class="field">`egress`</a> <span class="type">Map</span>  
```

File: site/content/docs/manifest/environment.en.md [hunk 7]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+
+<span class="parent-field">network.vpc.security-group.</span><a id="network-vpc-security-group-egress" href="#network-vpc-security-group" class="field">`egress`</a> <span class="type">Array of Security Group Rules</span>    
+A list of outbound security group rules.
+
+
+<span class="parent-field">network.vpc.security_group.<type\>.</span><a id="network-vpc-security-group-ip-protocol" href="#network-vpc-security-group-ip-protocol" class="field">`ip_protocol`</a> <span class="type">String</span>    
+The IP protocol name.
+
+<span class="parent-field">network.vpc.security_group.<type\>.</span><a id="network-vpc-security-group-ports" href="#network-vpc-security-group-ports" class="field">`ports`</a> <span class="type">String or Integer</span>     
+You can specify FromPort and ToPort for an environment security group rule.
```
Comment (Lou1415926) [inline site/content/docs/manifest/environment.en.md]: `FromPort` and `ToPort` are the CFN field name and my concern is that they seem to appear here out of context. What you do think of
```suggestion
The port range for the security group rule. 
```

File: site/content/docs/manifest/environment.en.md [hunk 8]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+```
+<span class="parent-field">network.vpc.security_group.</span><a id="network-vpc-security-group-ingress" href="#network-vpc-security-group" class="field">`ingress`</a> <span class="type">Array of Security Group Rules</span>    
+A list of inbound security group rules.
+
+<span class="parent-field">network.vpc.security-group.</span><a id="network-vpc-security-group-egress" href="#network-vpc-security-group" class="field">`egress`</a> <span class="type">Array of Security Group Rules</span>    
+A list of outbound security group rules.
+
+
+<span class="parent-field">network.vpc.security_group.<type\>.</span><a id="network-vpc-security-group-ip-protocol" href="#network-vpc-security-group-ip-protocol" class="field">`ip_protocol`</a> <span class="type">String</span>    
+The IP protocol name.
```
Comment (Lou1415926) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
The IP protocol name or number.
```
I *think* the protocol numbers also works: https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml

File: site/content/docs/manifest/environment.en.md [hunk 9]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+```
+
+Alternatively, you can specify it as an Integer if both FromPort and ToPort is same
+
+```yaml
+ports: n
+```
+
+<span class="parent-field">network.vpc.security_group.<type\>.</span><a id="network-vpc-security-group-cidr" href="#network-vpc-security-group-cidr" class="field">`cidr`</a> <span class="type">String</span>   
+a CIDR range.
```
Comment (Lou1415926) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
The IPv4 address range, in CIDR format.
```

**Review by Lou1415926 — APPROVED** [review]
lgtm! just some nits. feel free to remove the label when ur ready!

File: site/content/docs/manifest/environment.en.md [hunk 10]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
 The Availability Zone name assigned to the subnet. The `az` field is optional, by default Availability Zones are assigned in alphabetical order.
 This field is mutually exclusive with `id`.
 
+<span class="parent-field">network.vpc.</span><a id="network-vpc-security-group" href="#network-vpc-security-group" class="field">`security_group`</a> <span class="type">Map</span>  
+Rules of the environment's security group.
```
Comment (dannyrandall) [inline site/content/docs/manifest/environment.en.md]:
```suggestion
Rules for the environment's security group.
```

File: site/content/docs/manifest/environment.en.md [hunk 11]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+
+
+<span class="parent-field">network.vpc.security_group.<type\>.</span><a id="network-vpc-security-group-ip-protocol" href="#network-vpc-security-group-ip-protocol" class="field">`ip_protocol`</a> <span class="type">String</span>    
+The IP protocol name.
+
+<span class="parent-field">network.vpc.security_group.<type\>.</span><a id="network-vpc-security-group-ports" href="#network-vpc-security-group-ports" class="field">`ports`</a> <span class="type">String or Integer</span>     
+You can specify FromPort and ToPort for an environment security group rule.
+
+```yaml
+ports: n-m
```
Comment (dannyrandall) [inline site/content/docs/manifest/environment.en.md]: maybe just do a concrete example? (same below)
```suggestion
ports: 0-1023
```

File: site/content/docs/manifest/environment.en.md (hunk 7 above)
Comment (dannyrandall) [reply site/content/docs/manifest/environment.en.md]: maybe just say `The port range or number for the security group rule.`? and change the section  `Alternatively, you can specify it as an Integer if both FromPort and ToPort is same` to just `or`?

File: site/content/docs/manifest/environment.en.md (hunk 8 above)
Comment (paragbhingre) [reply site/content/docs/manifest/environment.en.md]: Right they work as numbers too. I will add number too

File: site/content/docs/manifest/environment.en.md [hunk 12]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+<span class="parent-field">network.vpc.security_group.</span><a id="network-vpc-security-group-egress" href="#network-vpc-security-group-egress" class="field">`egress`</a> <span class="type">Map</span>  
+Rules controlling outbound traffic.
+```yaml
+network:
+  vpc:
+    security_group:
+      egress:
+        - ip_protocol: tcp
+          ports: 80
+          cidr: 0.0.0.0/0
```
Comment (dannyrandall) [inline site/content/docs/manifest/environment.en.md]: small nit: let's flip the port ranges between egress and ingress. `ingress` to `80` and `egress` to `0-65535`. i think those are more realistic configurations

**Review by dannyrandall — APPROVED** [review]

File: site/content/docs/manifest/environment.en.md (hunk 12 above)
Comment (paragbhingre) [reply site/content/docs/manifest/environment.en.md]: Ohh you are right

File: site/content/docs/manifest/environment.en.md [hunk 13]
```diff
@@ -145,6 +145,59 @@ An IPv4 CIDR block assigned to the subnet. This field is mutually exclusive with
...
+```
+<span class="parent-field">network.vpc.security_group.</span><a id="network-vpc-security-group-ingress" href="#network-vpc-security-group" class="field">`ingress`</a> <span class="type">Array of Security Group Rules</span>    
+A list of inbound security group rules.
+
+<span class="parent-field">network.vpc.security-group.</span><a id="network-vpc-security-group-egress" href="#network-vpc-security-group" class="field">`egress`</a> <span class="type">Array of Security Group Rules</span>    
+A list of outbound security group rules.
```
Comment (rclinard-amzn) [inline site/content/docs/manifest/environment.en.md]: This seems to be duplicated / conflicting with the previous lines?

Comment (paragbhingre) [reply site/content/docs/manifest/environment.en.md]: Addressed

**Review by rclinard-amzn — APPROVED** [review]
Great work, looks good to me!

**Pull request merged**